apis/v1alpha2/shared_types.go

```diff
@@ -383,3 +383,54 @@ type AnnotationKey string
 // +kubebuilder:validation:MinLength=0
 // +kubebuilder:validation:MaxLength=4096
 type AnnotationValue string
+
+// TrafficMatches defines AddressMatch rules for inbound traffic according to
+// source and/or destination of that traffic.
+type TrafficMatches struct {
```

Comment:
nit: I find `TrafficMatches` a bit too vague as a type name here but I can't seem to come up with a better one. This is not a very big deal since it doesn't show up in the YAML API.

Reply:
This struct defines a route match, so we can follow the pattern in HTTPRoute and call it `AddressRouteMatch`, maybe with aliases `TCPRouteMatch` and `UDPRouteMatch`.

Reply:
I'm not opposed to calling this `AddressRouteMatches` if that suits you as well @hbagdi

Reply:
That's an improvement so +1.

Reply:
Alright sounds good, 19f1646b93f7db2e095771848c30fa1c7088c8ea changes this and updates the GEP accordingly.